syft/execution/type_wrapper.py

```diff
@@ -0,0 +1,261 @@
+import syft as sy
+from syft.workers.abstract import AbstractWorker
+from warnings import warn
+from typing import Union
+from syft_proto.execution.v1.type_wrapper_pb2 import NestedTypeWrapper as NestedTypeWrapperPB
+from syft_proto.execution.v1.type_wrapper_pb2 import ClassType as ClassTypePB
+
+class NestedTypeWrapper:
+    """
+        Class for input type serialization and type checking for nested structures.
+    """
+    def __init__(self, nested_type=None):
+        if nested_type:
+            self.serialized_nested_type = NestedTypeWrapper.serialize_nested_type(nested_type)
+        else:
+            self.serialized_nested_type = None
+
+    @staticmethod
+    def serialize_nested_type(input_arg: any) -> Union[list, tuple, dict, type]:
```

Comment:
I don't think this is exactly serialization (which would be the responsibility of the serdes.) Maybe it's something like `enumerate_nested_types`?